apps/search/documents.py

```diff
@@ -0,0 +1,125 @@
+from django_elasticsearch_dsl import Document, fields
+from django_elasticsearch_dsl.registries import registry
+from elasticsearch_dsl import analyzer
+
+from apps.occurrences.models import Occurrence
+from apps.quotes.models import Quote
+from apps.entities.models import Entity
+from apps.sources.models import Source
+from apps.images.models import Image
+
+html_strip = analyzer(
+    'html_strip',
+    tokenizer="standard",
+    filter=["lowercase", "stop", "snowball"],
+    char_filter=["html_strip"]
+)
+
+DEFAULT_INDEX_SETTINGS = {'number_of_shards': 1, 'number_of_replicas': 0}
+
+
+@registry.register_document
+class QuoteDocument(Document):
+    class Index:
+        settings = DEFAULT_INDEX_SETTINGS
+        name = 'quotes'
+
+    text = fields.TextField(attr='text.text', analyzer=html_strip)
+    context = fields.TextField(attr='context.text', analyzer=html_strip)
+    date = fields.TextField(attr='date.string')
+    attributees = fields.ObjectField(properties={
+        'id': fields.IntegerField(),
+        'name': fields.TextField(),
+        'aliases': fields.TextField(),
+        'description': fields.TextField(attr='description.text')
+    })
+    citations = fields.TextField(attr='citation_html', analyzer=html_strip)
+    topics = fields.ObjectField(attr='_related_topics', properties={
+        'id': fields.IntegerField(),
+        'key': fields.TextField(),
+        'aliases': fields.TextField(),
+        'description': fields.TextField(attr='description.text', analyzer=html_strip)
+    })
+
+    class Django:
+        model = Quote
+        related_models = [Source, Entity]
+
+    def get_queryset(self):
+        return super(QuoteDocument, self).get_queryset().prefetch_related(
+            'attributees', 'sources', 'topics'
+        )
+
+
+@registry.register_document
+class OccurrenceDocument(Document):
+    class Index:
+        settings = DEFAULT_INDEX_SETTINGS
+        name = 'occurrences'
+
+    summary = fields.TextField(attr='summary.text', analyzer=html_strip)
+    description = fields.TextField(attr='description.text', analyzer=html_strip)
+    date = fields.TextField(attr='date.string')
+    involved_entities = fields.ObjectField(properties={
+        'id': fields.IntegerField(),
+        'name': fields.TextField(),
+        'aliases': fields.TextField(),
+        'description': fields.TextField(attr='description.text')
+    })
+    citations = fields.TextField(attr='citation_html', analyzer=html_strip)
+    topics = fields.ObjectField(attr='_related_topics', properties={
+        'id': fields.IntegerField(),
+        'key': fields.TextField(),
+        'aliases': fields.TextField(),
+        'description': fields.TextField(attr='description.text', analyzer=html_strip)
+    })
+
+    class Django:
+        model = Occurrence
+        related_models = [Source, Entity]
+
+    def get_queryset(self):
+        return super(OccurrenceDocument, self).get_queryset().prefetch_related(
```

Comment:
nitpick: We can use `super().get_queryset()` rather than `super(OccurrenceDocument, self).get_queryset()`